cmd/axelard/cmd/vald/start.go

```diff
@@ -320,30 +319,53 @@ func createJob(sub tmEvents.FilteredSubscriber, processor func(event tmEvents.Ev
 
 }
 
-func getStartBlock(stateStore StateStore, tmClient tmEvents.BlockHeightClient, logger log.Logger) (int64, error) {
-	startBlock, err := stateStore.GetState()
-	if err != nil {
-		logger.Error(err.Error())
-
-		return 0, nil
-	}
-
+// Wait until the node has synced with the network
+// and then return the block height to start listening to TM events from
+func waitTillNetworkSync(cfg config.ValdConfig, stateStore StateStore, tmClient tmEvents.SyncInfoClient, logger log.Logger) (int64, error) {
 	rpcCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
-	height, err := tmClient.LatestBlockHeight(rpcCtx)
+	syncInfo, err := tmClient.LatestSyncInfo(rpcCtx)
 	if err != nil {
 		return 0, err
 	}
 
-	// TODO: Make it configurable instead of a hardcoded 100
-	if height-startBlock > 100 {
-		logger.Info(fmt.Sprintf("block in state %d is too old and will start from the latest instead", startBlock))
+	cachedHeight, err := stateStore.GetState()
+	if err != nil {
+		logger.Info(fmt.Sprintf("failed to retrieve the cached block height, using the latest: %s", err.Error()))
+		cachedHeight = 0
+	} else {
+		logger.Info(fmt.Sprintf("retrieved cached block height %d", cachedHeight))
+		cachedHeight++ // Skip the block that might have already been executed
+	}
+
+	// cached height must not be more than one block ahead of the node
+	if cachedHeight > syncInfo.LatestBlockHeight+1 {
+		return 0, fmt.Errorf("cached block height %d is ahead of the node height %d", cachedHeight, syncInfo.LatestBlockHeight)
+	}
+
+	// If the block height is older than the allowed time, then wait for the node to sync
+	for syncInfo.LatestBlockTime.Add(cfg.MaxLatestBlockAge).Before(time.Now()) {
+		logger.Info(fmt.Sprintf("node height %d is old, waiting for a recent block", syncInfo.LatestBlockHeight))
+		time.Sleep(cfg.MaxLatestBlockAge)
```

Comment:
```suggestion
		time.Sleep(cfg.MaxLatestBlockAge)/2
```
If you wait the full duration by the time you check the condition the block might be expired again

Reply:
This is why the `MaxLatestBlockAge` should be set to a value larger than any realistic block production time. For e.g. the default is set to `15s`. If block times are around 6s, this should never be an issue. If block times increase by that much, then we can change the default or ask validators to update it, since this is only a problem on vald start-up.